windows/win_host.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2014, Phil Schwartz <schwartzmx@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# this is a windows documentation stub.  actual code lives in the .ps1
+# file of the same name
+
+DOCUMENTATION = '''
+---
+module: win_host
+version_added: ""
+short_description: Hostname setting, and domain unjoining/joining module.
+description:
+     - Change the host's hostname, and domain.  Allows active-directory domain joining/unjoining, along with workgroup joining/unjoining. Uses Powershell's Add-Computer and Remove-Computer.
+options:
+  hostname:
+    description:
+      - Hostname to change to, or a comma separated list of computers to add to the domain
+    required: no
+    default: Current Host Name
+    aliases: []
+  state:
+    description:
+      - Specify whether to join (present) or unjoin (absent)
+    required: no
+    choices:
+      - present
+      - absent
+    default: none
+    aliases: []
+  domain:
+    description:
+      - Domain name to join/unjoin
+    required: no
+    default: none
+    aliases: []
+  rm:
+    description:
+      - *Specified when the play acts on a DOMAIN CONTROLLER, it will remove the given hostname from the ActiveDirectory Computer users.
+    required: no
+    default: none
+    aliases: []
+  workgroup:
+    description:
+      - Workgroup name to join/unjoin
+    required: no
+    default: WORKGROUP
+    aliases: []
+  server:
+    description:
+      - Name of a domain controller
+    required: no
+    default: none
+    aliases: []
+  user:
+    description:
+      - Domain user with permission to join/unjoin
+    required: no
+    default: none
+    aliases: []
+  pass:
+    description:
+      - Domain user password
+    required: no
+    default: none
+    aliases: []
+  options:
+    description:
+      - Single or comma separated list of options (AccountCreate, Win9XUpgrade, UnsecuredJoin, PasswordPass, JoinWithNewName, JoinReadOnly, InstallInvoke)
+    required: no
+    default: none
+    aliases: []
+  oupath:
+    description:
+      - Specifies an organizational unit for the domain account
+    required: no
+    default: none
+    aliases: []
+  restart:
+    description:
+      - Restart the host after completion
+    required: no
+    choices:
+      - true
+      - yes
+      - false
+      - no
+    default: false
```

Comment:
@cchurch noted that I should specify these on another module PR [here](https://github.com/ansible/ansible-modules-core/pull/1109#issuecomment-140284424),  not sure what the consensus is.